pkg/keycloak/jsonInjector.go

```diff
@@ -0,0 +1,166 @@
+package keycloak
+
+import (
+	"fmt"
+	"github.com/Jeffail/gabs"
+)
+
+// JsonInjector provides functions to inject kubernetes resources into a JSON
+// template
+type JsonInjector interface {
+	ParseTemplate(template []byte) (*gabs.Container, error)
+	InjectEnvVar(*gabs.Container) (*gabs.Container, error)
+	InjectVolume(template []byte) ([]byte, error)
+	InjectVolumeMount(template []byte) ([]byte, error)
+	InjectInitContainer(template []byte) ([]byte, error)
+	InjectAll(template []byte) ([]byte, error)
+}
+
+// JsonInjectorImpl contains variables to be replaced by the injector
+type JsonInjectorImpl struct {
+	PluginsEnvVarName  string
+	PluginsVolumeName  string
+	PluginVolumeMount  string
+	InitContainerName  string
+	InitContainerMount string
+	InitContainerImage string
+}
+
+// Create a new JsonInjector
+func newJsonInjector() *JsonInjectorImpl {
+	return &JsonInjectorImpl{
+		PluginsEnvVarName:  "SSO_PLUGINS",
+		PluginsVolumeName:  "sso-plugins",
+		PluginVolumeMount:  "/opt/eap/providers",
+		InitContainerName:  "sso-plugins-init",
+		InitContainerMount: "/opt/plugins",
+		InitContainerImage: "docker.io/pb82/kc_plugins_init:latest",
```

Comment:
Should we push this image to the `integr8ly` `quay.io` registry and reference that instead?